### ISSUE TITLE
Allow to include references to keys in headers

### ISSUE THREAD

**Issue by azul** [mailencrypt/inbome#1]
Use Case: Journalist / Lawyer / human rights org knows people who contact them might want to introduce themselves in an encrypted message. They want to be able to spread their encryption keys widely so they can be contacted with an encrypted email.

There's an old standard for this: https://www.ietf.org/archive/id/draft-josefsson-openpgp-mailnews-header-07.txt
But it is hard to parse by the MUA in a lot of the examples. Let's have a machine readable way of specifying your key as a sender without including it. This could be done by fingerprint if the key is on keyservers or by url.

**Comment by azul** [comment]
Alternatives:
1. **Signing emails**: Turns out people are confused by the requirement of typing in the password if you sign every mail by default (according to k9 devs).
2. **Attaching key**: Keys can be fairly large these days. If needed we have already have the inbome provide header that can include key data.

In addition specifying a reference to the key also allows updating the key via that reference. So if we exchanged some mails a while ago, your key expired and you specified a url to retrieve it from or a fingerprint i could use those to try and update the key.

Making use of this information obviously requires out of band mechanisms to retrieve the keys themselves such as keyserver key lookup or https requests. These requests are thus not part of the imbome spec. But we still enable them.

**Comment by hpk42** [comment]
I don't quite understand the english in the use case description to be honest.  Why could those journalists etc not just send along their actual encryption key in mails and be done?

**Comment by azul** [comment]
because those mails might go to mailing lists etc. They don't know who will contact them. The idea is to spread the info far and wide.
But yes. Just attaching the key to all mails they send might also be an option. Extra attachments also tend to be confusing. Does the spec say something about using and hiding attached keys?

**Comment by hpk42** [comment]
inbome-specs as of now do not discuss, specialize wrt mailnig lists.  So mailing lists could decide to scrap key headers.  INBOME does not attach keys as things stand, btw.  So for now, i don't see how the use case you describe makes the ability to reference neccessary. 

i think a potentially good reason we may want to add the ability to reference instead of send a key verbatim is compatibility with existing pgp keyserver/infrastructure.  Particularly mailvelope keyservers operated by providers are interesting. 

I know that @Valodim also considers using keyservers for handling revocation of INBOME keys.

**Comment by azul** [comment]
Let me try and rephrase this. What I would like to enable in some cases is encryption of the second mail rather than the third. From my point of view there is a tradeoff involved here between key availability and noise (message size, complexity, ui issues with clients that do not support inbome).

Best key availability: Include the key in every message. All recipients can reply in an encrypted way.
Lowest noise: only announce capability.

These two options exist within the spec as far as i can see. For me there would be a middleground that depends on out of band key retrieval. It's including a referrer to the key in a (well standardized) header. Either a fingerprint of a key available on keyservers or a url.

This would create little noise and still enable inbome capable clients that are fine with using out of band means to retrieve the key.

**Comment by azul** [comment]
Just noticed that the standard now already encourages including key data and thus moves encryption one step ahead. :sparkles:  That was what mattered to me. Closing this.
